### PR TITLE
chore: rename DefinitionRoot -> Root

### PR DESCRIPTION
**Pull request by tatomyr**
## What/Why/How?

Closes https://github.com/Redocly/redocly-cli/issues/839

Renamed also other types:
```
  Root: 'DefinitionRoot',
  ServerVariablesMap: 'ServerVariableMap',
  PathsMap: 'PathMap',
  CallbacksMap: 'CallbackMap',
  MediaTypesMap: 'MediaTypeMap',
  ExamplesMap: 'ExampleMap',
  EncodingsMap: 'EncodingMap',
  HeadersMap: 'HeaderMap',
  LinksMap: 'LinkMap',
```

## Check yourself

- [x] Code is linted
- [x] Tested with redoc/reference-docs/workflows
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

**Review by adamaltman — COMMENTED** [review]
I like it but I think we should consider adding that `DefinitionRoot` is defined as type `Root` for some period of backwards compatibility.

**Review by adamaltman — APPROVED** [review]
LGTM

File: packages/core/src/visitors.ts [hunk 1]
```diff
@@ -380,8 +393,11 @@ export function normalizeVisitors<T extends BaseVisitor>(
     }
 
     for (const typeName of visitorKeys as Array<keyof T>) {
-      const typeVisitor = visitor[typeName] as any as NestedVisitObject<any, T>;
-      const normalizedTypeVisitor = normalizedVisitors[typeName]!;
+      const typeVisitor = (visitor[typeName] ||
```
Comment (RomanHotsiy) [inline packages/core/src/visitors.ts:396]: do we have a test for a legacy names somewhere? if not we should add temporary.

**Review by RomanHotsiy — APPROVED** [review]
LGTM

File: packages/core/src/visitors.ts (hunk 1 above)
Comment (tatomyr) [reply packages/core/src/visitors.ts:396]: Good point. Will add a them.

**Pull request merged**